lib/keypair.rb

```diff
@@ -229,4 +231,33 @@ def expires_at_after_not_after
 
     errors.add(:expires_at, 'must be after not after')
   end
+
+  class << self
+    private
+
+    LOCK_NAME = 'keypairs:rotate'
+
+    def rotate_keys_with_lock!
+      with_advisory_lock(LOCK_NAME) do
```

Comment:
Hmm, Isn't this a bit overkill? How does this relate with the pessimistic locking like https://api.rubyonrails.org/classes/ActiveRecord/Locking/Pessimistic.html

Reply:
I agree with you that a solution like pessimistic or optimistic locking would be easier. However, we don't always have a clear idea which record to use for locking. For example, when we don't have any keypairs yet we don't have any records which we can lock. It's also not clear on which record we should lock, especially when there are multiple keypairs for the same time period, as there would be when migrating from a previous version of the gem.